ci/Jenkinsfiles/build.groovy

```diff
@@ -0,0 +1,445 @@
+                    envsubst < ci/mvn/nuxeo-test-${env}.properties~gen > ${HOME}/nuxeo-test-${env}.properties
+                """
+              }
+              // run unit tests for the given environment (see the customEnvironment profile in pom.xml):
+              //   - in an alternative build directory
+              //   - loading some test framework system properties
+              def testCore = env == 'mongodb' ? 'mongodb' : 'vcs'
+              def kafkaOptions = isDev ? '' : "-Pkafka -Dkafka.bootstrap.servers=${kafkaHost}"
+              def mvnCommand = """                 
+                mvn ${MAVEN_ARGS} \
+                  -Dcustom.environment=${env} \
+                  -Dcustom.environment.log.dir=target-${env} \
+                  -Dnuxeo.test.core=${testCore} \
+                  ${kafkaOptions} \
+                  test
+              """
+              sh "${mvnCommand}"
+
+              setGitHubBuildStatus("utests/${env}", "Unit tests - ${env} environment", 'SUCCESS')
+            } catch (err) {
+              echo "${env} unit tests error: ${err}"
+              setGitHubBuildStatus("utests/${env}", "Unit tests - ${env} environment", 'FAILURE')
+              throw err
+            } finally {
+              try {
+                junit allowEmptyResults: true, testResults: "**/target-${env}/surefire-reports/*.xml"
+                if (!isDev) {
+                  archiveKafkaLogs(testNamespace, "${env}-kafka.log")
+                }
+              } finally {
+                echo "${env} unit tests: clean up test namespace"
+                try {
+                  if(!isDev){
+                    helmfileDestroy("${testNamespace}", "${environment}")
+                  }
+                } finally {
+                  // clean up test namespace
+                  sh "kubectl delete namespace ${testNamespace} --ignore-not-found=true"
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+}
+
+String getCurrentNamespace() {
+  container('maven') {
+    return sh(returnStdout: true, script: "kubectl get pod ${NODE_NAME} -ojsonpath='{..namespace}'")
+  }
+}
+
+void archiveKafkaLogs(namespace, logFile) {
+  // don't fail if pod doesn't exist
+  sh "kubectl logs ${TEST_KAFKA_POD_NAME} --namespace=${namespace} > ${logFile} || true"
+  archiveArtifacts allowEmptyArchive: true, artifacts: "${logFile}"
+}
+
+void gitCheckout(String reference, String option = '') {
+  sh """
+    git checkout ${option} ${reference}
+  """
+}
+
+void gitPush(String reference) {
+  sh """
+    git push origin ${reference}
+  """
+}
+
+void gitTag(String tagname, String message) {
+  sh """
+    git tag -a ${tagname} -m "${message}"
+  """
+}
+
+pipeline {
+  agent {
+    label 'jenkins-nuxeo-platform-lts-2021'
+  }
+  options {
+    disableConcurrentBuilds()
+    buildDiscarder(logRotator(daysToKeepStr: '15', numToKeepStr: '10', artifactNumToKeepStr: '5'))
```

Comment:
As in https://github.com/nuxeo/nuxeo-lts/blob/2021/ci/Jenkinsfiles/build.groovy#L41., we usually use:
- daysToKeepStr: 60 # days to keep builds
- numToKeepStr: 60 # max numbers of builds to keep
- artifactNumToKeepStr: 5 # number of builds that have their artifacts kept

Let's keep in sync.